include/devicetree/clocks.h

```diff
@@ -22,6 +24,92 @@ extern "C" {
  * @{
  */
 
+/**
+ * @brief Test if a node has a clocks phandle-array property at a given index
+ *
+ * This expands to 1 if the given index is valid clocks property phandle-array index.
+ * Otherwise, it expands to 0.
+ *
+ * Example devicetree fragment:
+ *
+ *     n1: node-1 {
+ *             clocks = <...>, <...>;
+ *     };
+ *
+ *     n2: node-2 {
+ *             clocks = <...>;
+ *     };
+ *
+ * Example usage:
+ *
+ *     DT_CLOCKS_HAS_IDX(DT_NODELABEL(n1), 0) // 1
+ *     DT_CLOCKS_HAS_IDX(DT_NODELABEL(n1), 1) // 1
+ *     DT_CLOCKS_HAS_IDX(DT_NODELABEL(n1), 2) // 0
+ *     DT_CLOCKS_HAS_IDX(DT_NODELABEL(n2), 1) // 0
+ *
+ * @param node_id node identifier; may or may not have any clocks property
+ * @param idx index of a clocks property phandle-array whose existence to check
+ * @return 1 if the index exists, 0 otherwise
+ */
+#define DT_CLOCKS_HAS_IDX(node_id, idx) \
+	IS_ENABLED(DT_CAT4(node_id, _P_clocks_IDX_, idx, _EXISTS))
+
+/**
+ * @brief Test if a node has a clock-names array property holds a given name
+ *
+ * This expands to 1 if the name is available as clocks-name array property cell.
+ * Otherwise, it expands to 0.
+ *
+ * Example devicetree fragment:
+ *
+ *     n1: node-1 {
+ *             clocks = <...>, <...>;
+ *             clock-names = "alpha", "beta";
+ *     };
+ *
+ *     n2: node-2 {
+ *             clocks = <...>;
+ *             clock-names = "alpha";
+ *     };
+ *
+ * Example usage:
+ *
+ *     DT_CLOCKS_HAS_NAME(DT_NODELABEL(n1), alpha) // 1
+ *     DT_CLOCKS_HAS_NAME(DT_NODELABEL(n1), beta)  // 1
+ *     DT_CLOCKS_HAS_NAME(DT_NODELABEL(n2), beta)  // 0
+ *
+ * @param node_id node identifier; may or may not have any clock-names property.
+ * @param name lowercase-and-underscores clock-names cell value name to check
+ * @return 1 if the clock name exists, 0 otherwise
+ */
+#define DT_CLOCKS_HAS_NAME(node_id, name) \
+	IS_ENABLED(DT_CAT4(node_id, _P_clocks_NAME_, name, _EXISTS))
```

Comment:
Why not add a general-purpose helper for deciding if a named property exists, like `DT_PROP_HAS_NAME(node_id, prop, name)`, and using it here?

Reply:
Sure, will add `DT_PROP_HAS_NAME`